### PR TITLE
Cart delete functionality react

### PR DESCRIPTION
**Pull request by mayuresh-tech**

### CODE REVIEW

File: src/components/product-card/HorizontalCard/ProductCardHorizontal.jsx [hunk 1]
```diff
@@ -44,49 +44,153 @@ const ProductCardHorizontal = ({ item }) => {
...
+  }
+
+  async function removeFromCart(product) {
+    try {
+      const responseCart = await axios.get(`/api/user/cart`, {
+        headers: {
+          authorization: loginToken,
+        },
+      });
+      if (responseCart.data.cart.find((item) => item._id === product._id)) {
```
Comment (deekshithmd) [inline src/components/product-card/HorizontalCard/ProductCardHorizontal.jsx:113]: Suggestion: you can reduce one api call by using cart data in context.

File: src/components/product-card/HorizontalCard/ProductCardHorizontal.jsx [hunk 2]
```diff
@@ -44,49 +44,153 @@ const ProductCardHorizontal = ({ item }) => {
...
 
+  async function removeFromWishlist(product) {
+    try {
+      const responseWishlist = await axios.get(`/api/user/wishlist`, {
+        headers: {
+          authorization: loginToken,
+        },
+      });
+      if (
+        responseWishlist.data.wishlist.find((item) => item._id === product._id)
```
Comment (deekshithmd) [inline src/components/product-card/HorizontalCard/ProductCardHorizontal.jsx:55]: Same here, you can reduce one api call...

**Review by deekshithmd — APPROVED** [review]

File: src/components/product-card/HorizontalCard/ProductCardHorizontal.jsx (hunk 1 above)
Comment (mayuresh-tech) [reply src/components/product-card/HorizontalCard/ProductCardHorizontal.jsx:113]: I will fix these changes in next feature commit

File: src/components/product-card/HorizontalCard/ProductCardHorizontal.jsx (hunk 2 above)
Comment (mayuresh-tech) [reply src/components/product-card/HorizontalCard/ProductCardHorizontal.jsx:55]: I will fix these changes in next feature commit

**Pull request merged**